### PR TITLE
tests: add incremental mode test

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
add incremental mode test

### What is changed and how it works?
1. add test case, prepare data -> start task in full mode -> stop task -> incremental data -> start task in incremental mode
~2. remove useless code in checkpoint~ I will do it in another pr

### Check List 

Tests 

 - Integration test

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

File: tests/incremental_mode/run.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+    run_dm_worker $WORK_DIR/worker2 $WORKER2_PORT $cur/conf/dm-worker2.toml
+    run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml
+
+    check_port_alive $MASTER_PORT
+    check_port_alive $WORKER1_PORT
+    check_port_alive $WORKER2_PORT
+
+    cd $cur && GO111MODULE=on go build -o bin/dmctl dmctl.go && cd -
+    cd $cur && GO111MODULE=on go build -o bin/dmctl_stop dmctl_stop.go && cd -
```
Comment (csuzhangxc) [inline tests/incremental_mode/run.sh]: can we re-use these binaries between multiple test cases?

Comment (amyangfei) [reply tests/incremental_mode/run.sh]: added global dmctl tools, now each test case can use them without re-compiling

Comment (csuzhangxc) [reply tests/incremental_mode/run.sh]: do we need to add retry for _dm_start_task.go_?

Comment (amyangfei) [reply tests/incremental_mode/run.sh]: currently not needed, If we need retry mechanism we can wrap the binary from caller

**Comment by csuzhangxc** [comment]
LGTM

File: tests/dmctl_tools/dmctl_operate_task.go [hunk 2]
```diff
@@ -15,18 +15,28 @@ package main
...
+	if len(os.Args) != 3 {
+		utils.ExitWithError(fmt.Errorf("invlid args: %v", os.Args))
+	}
 	cli, err := utils.CreateDmCtl("127.0.0.1:8261")
 	if err != nil {
 		utils.ExitWithError(err)
 	}
-	conf := os.Args[1]
-	err = utils.StartTask(context.Background(), cli, conf, nil)
+	op, err := strconv.ParseInt(os.Args[1], 10, 64)
```
Comment (GregoryIan) [inline tests/dmctl_tools/dmctl_operate_task.go]: can we use `string` as type of  argumen `op`? and then convert `op` to `pb.TaskOp` in this function

File: tests/utils/dmctl.go [hunk 3]
```diff
@@ -58,3 +58,22 @@ func StartTask(ctx context.Context, cli pb.MasterClient, configFile string, work
 
 	return nil
 }
+
```
Comment (GregoryIan) [inline tests/utils/dmctl.go]: why not moving it to `dm_tools` directory?

File: tests/dmctl_tools/dmctl_operate_task.go (hunk 2 above)
Comment (amyangfei) [reply tests/dmctl_tools/dmctl_operate_task.go]: changed to use string type argument, feel free to covert to pb.TaskOp then

File: tests/utils/dmctl.go (hunk 3 above)
Comment (amyangfei) [reply tests/utils/dmctl.go]: we put binary entry in `dm_tools` directory, and `utils` directory behave more like pkg

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**